synapse_net/training/domain_adaptation.py

```diff
@@ -82,6 +84,10 @@ def mean_teacher_adaptation(
             based on the patch_shape and size of the volumes used for training.
         n_samples_val: The number of val samples per epoch. By default this will be estimated
             based on the patch_shape and size of the volumes used for validation.
+        train_mask_paths: Boundary masks used by the sampler to accept or reject patches for training. 
+        val_mask_paths: Boundary masks used by the sampler to accept or reject patches for     validation. 
+        sampler: Accept or reject patches based on a condition.
```

Comment:
The samplers for the datasets and the mean teacher trainer have slightly different meaning. See also comment below. I think the best approach here is to expose and document two different sampler arguments:
- `patch_sampler`: is passed to `get_unsupervised_loader`
- `pseudo_label_sampler`: is passed to `MeanTeacherTrainer`

Feel free to suggest better names ;).